### PR TITLE
CIRCSTORE-426 Update request policies if ServicePoint is updated

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
- Create a handler for handling "service point update" events. If the service point's pickupLocation field was changed from Yes (true) to No (false), it should find all of the request policies that reference the updated service point and remove it from there

Resolves: [CIRCSTORE-426](https://issues.folio.org/browse/CIRCSTORE-426)

## Approach
- add logic for removing allowed SPs from request policies if pickup location is changed from true to false;
- add tests;

### CODE REVIEW

File: src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        .setJSONB(true)
+        .setVal(String.format("[\"%s\"]", servicePointId)))
+      .toList();
+    GroupedCriterias groupedCriterias = new GroupedCriterias();
+    criteriaList.forEach(criteria -> groupedCriterias.addCriteria(criteria, "OR"));
+
+    return policyRepository.get(new Criterion().addGroupOfCriterias(groupedCriterias));
+  }
+
+  private Future<List<RequestPolicy>> removeAllowedServicePoints(RequestPolicyRepository policyRepository,
```
Comment (alexanderkurash) [inline src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java:94]: Remove

File: src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      log.info("removeServicePointId:: removing allowed servicePointId from Page");
+      pageAllowedServicePoints.remove(servicePointId);
+    }
+
+    Set<String> recallAllowedServicePoints = allowedServicePoints.getRecall();
+    if (recallAllowedServicePoints != null && !recallAllowedServicePoints.isEmpty()) {
+      log.info("removeServicePointId:: removing allowed servicePointId from Recall");
+      recallAllowedServicePoints.remove(servicePointId);
+    }
+    log.info("removeServicePointId:: result: {}", policy);
```
Comment (alexanderkurash) [inline src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java:131]: Delete request type when 0 service points left
Delete allowedServicePoints when 0 types left

File: src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java (hunk 1 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java]: done

File: src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java (hunk 2 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java]: done

File: src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  protected List<Change<RequestPolicy>> collectRelevantChanges(JsonObject payload) {
+    JsonObject oldObject = payload.getJsonObject("old");
+    JsonObject newObject = payload.getJsonObject("new");
+    List<Change<RequestPolicy>> changes = new ArrayList<>();
+    String updatedServicePointId = oldObject.getString("id");
+
+    Boolean isOldServicePointPickupLocation = oldObject.getBoolean(SERVICE_POINT_PICKUP_LOCATION);
+    Boolean isNewServicePointPickupLocation = newObject.getBoolean(SERVICE_POINT_PICKUP_LOCATION);
+    if (isOldServicePointPickupLocation != null && isNewServicePointPickupLocation != null &&
+      !isNewServicePointPickupLocation && isOldServicePointPickupLocation) {
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java:38]:
```suggestion
    if (isOldServicePointPickupLocation == TRUE && isNewServicePointPickupLocation == FALSE) {
```

Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/service/event/handler/processor/ServicePointUpdateProcessorForRequestPolicy.java:38]: Those are `Boolean.TRUE` and `Boolean.FALSE`

**Pull request merged**